src/mapgen/mg_biome.cpp

```diff
@@ -147,18 +149,51 @@ BiomeGenOriginal::BiomeGenOriginal(BiomeManager *biomemgr,
 	// fallback biome when biome generation (which calculates the biomemap IDs)
 	// is disabled.
 	memset(biomemap, 0, sizeof(biome_t) * m_csize.X * m_csize.Z);
+
+	// Calculating the bounding position of each biome so we know when we might switch
+	biomeTransitions = new s16[m_bmgr->getNumObjects() * 2];
```

Comment:
Please use `snake_case` variable names for consistency: https://dev.minetest.net/Code_style_guidelines#Miscellaneous